### PR TITLE
Some entity mappings

### PR DESCRIPTION
**Pull request by Fuyukai**
Was unsure what to name the death screen functions; was torn between showDeathScreen and notInstantRespawn but the latter sounds kind of weird and this is only used on the client for is the death screen should be shown.

Doesn't clash with #882.

### CODE REVIEW

**Comment by Fuyukai** [comment]
Whoops, didn't mean to do that

**Comment by liach** [comment]
@Fuyukai just change the base branch of this pr to 19w40a

**Comment by Fuyukai** [comment]
I figured a git rebase would do that automatically on github's side, but I guess not.

File: mappings/net/minecraft/server/network/packet/ClientSettingsC2SPacket.mapping [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 		ARG 1 language
 		ARG 2 viewDistance
 		ARG 3 chatVisibility
+		ARG 4 chatColors
 		ARG 5 modelBitMask
 		ARG 6 mainArm
 	METHOD method_12131 getLanguage ()Ljava/lang/String;
 	METHOD method_12132 getMainArm ()Lnet/minecraft/class_1306;
 	METHOD method_12134 getChatVisibility ()Lnet/minecraft/class_1659;
+	METHOD method_12135 getChatColors ()Z
```
Comment (liach) [inline mappings/net/minecraft/server/network/packet/ClientSettingsC2SPacket.mapping]: No `getXxx` for boolean getters

**Review by liach — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/server/network/packet/ClientSettingsC2SPacket.mapping (hunk 1 above)
Comment (Pyrofab) [reply mappings/net/minecraft/server/network/packet/ClientSettingsC2SPacket.mapping]: `areChatColorsEnabled` may be a good alternative

Comment (liach) [reply mappings/net/minecraft/server/network/packet/ClientSettingsC2SPacket.mapping]: `hasChatColors` or `allowsChatColors`
```suggestion
	METHOD method_12135 hasChatColors ()Z
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+	FIELD field_7655 yVelocityClient D
+	FIELD field_7656 zVelocityClient D
+	FIELD field_7657 pitchClient D
+	FIELD field_7658 xVelocityClient D
+	FIELD field_7659 yawClient D
 	FIELD field_7661 CUSTOM_BLOCK_OFFSET Lnet/minecraft/class_2940;
+	FIELD field_7662 zPosClient D
 	FIELD field_7663 DAMAGE_WOBBLE_TICKS Lnet/minecraft/class_2940;
+	FIELD field_7665 xPosClient D
+	FIELD field_7666 yPosClient D
```
Comment (Pyrofab) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]: These should probably reflect the regular position fields and drop the `pos` part. (xClient)

Comment (liach) [reply mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]: Yeah I actually disagree with these minecart field mappings as well

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 3]
```diff
@@ -1,9 +1,18 @@
 CLASS net/minecraft/class_1688 net/minecraft/entity/vehicle/AbstractMinecartEntity
 	CLASS class_1689 Type
+	FIELD field_7655 yVelocityClient D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7655 clientVelocityY D
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
+	FIELD field_7656 zVelocityClient D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7656 clientVelocityZ D
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 5]
```diff
@@ -1,9 +1,18 @@
...
+	FIELD field_7657 pitchClient D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7657 clientPitch D
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 6]
```diff
@@ -1,9 +1,18 @@
...
+	FIELD field_7658 xVelocityClient D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7658 clientVelocityX D
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 7]
```diff
@@ -1,9 +1,18 @@
...
+	FIELD field_7659 yawClient D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7659 clientYaw D
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 8]
```diff
@@ -1,9 +1,18 @@
...
 	FIELD field_7661 CUSTOM_BLOCK_OFFSET Lnet/minecraft/class_2940;
+	FIELD field_7662 zPosClient D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7662 clientZ D
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 9]
```diff
@@ -1,9 +1,18 @@
...
 	FIELD field_7663 DAMAGE_WOBBLE_TICKS Lnet/minecraft/class_2940;
+	FIELD field_7665 xPosClient D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7665 clientX D
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping (hunk 2 above)
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7666 clientY D
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 10]
```diff
@@ -1,9 +1,18 @@
...
+	FIELD field_7666 yPosClient D
 	FIELD field_7667 DAMAGE_WOBBLE_STRENGTH Lnet/minecraft/class_2940;
 	FIELD field_7668 DAMAGE_WOBBLE_SIDE Lnet/minecraft/class_2940;
+	FIELD field_7669 interpolationStepsClient I
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	FIELD field_7669 clientInterpolationSteps I
```

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 11]
```diff
@@ -14,15 +23,23 @@ CLASS net/minecraft/class_1688 net/minecraft/entity/vehicle/AbstractMinecartEnti
 		ARG 3 x
 		ARG 5 y
 		ARG 7 z
+	METHOD method_18803 willHitBlockAt (Lnet/minecraft/class_2338;)Z
+		ARG 1 pos
+	METHOD method_7504 getMaxSpeedOffRail ()D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]:
```suggestion
	METHOD method_7504 getMaxOffRailSpeed ()D
```

**Review by liach — CHANGES_REQUESTED** [review]
Just fix the name order. We say thing like `client velocity x` (client velocity's x component) not `x velocity client`.

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping (hunk 11 above)
Comment (Fuyukai) [reply mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]: I would actually say "get max speed off the rail" in normal speech, not "get max off rail speed".

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
+	FIELD field_7658 clientXVelocity D
+	FIELD field_7659 clientYaw D
 	FIELD field_7661 CUSTOM_BLOCK_OFFSET Lnet/minecraft/class_2940;
-	FIELD field_7662 zPosClient D
+	FIELD field_7662 clientZPos D
 	FIELD field_7663 DAMAGE_WOBBLE_TICKS Lnet/minecraft/class_2940;
-	FIELD field_7665 xPosClient D
-	FIELD field_7666 yPosClient D
+	FIELD field_7665 clientXPos D
+	FIELD field_7666 clientYPos D
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]: Unnecessary `Pos` suffix

File: mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping [hunk 13]
```diff
@@ -1,18 +1,18 @@
...
 	FIELD field_7667 DAMAGE_WOBBLE_STRENGTH Lnet/minecraft/class_2940;
 	FIELD field_7668 DAMAGE_WOBBLE_SIDE Lnet/minecraft/class_2940;
-	FIELD field_7669 interpolationStepsClient I
+	FIELD field_7669 cliientInterpolationSteps I
```
Comment (liach) [inline mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping]: extra `i` for `client`

**Review by liach — CHANGES_REQUESTED** [review]

**Comment by liach** [comment]
Please updating the target version (rebase against 19w41a branch and change base branch of this pr) and resolving conflicts.

**Comment by Fuyukai** [comment]
Sorry about the long delays between fixes and bunch of mistakes - I haven't been up and at my desk much the last few weeks >.>

**Review by liach — APPROVED** [review]

**Comment by liach** [comment]
Please change base branch to 19w42a.

**Review by YanisBft — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**